src/tplr/comms.py

```diff
@@ -1528,6 +1528,9 @@ async def get_peer_list(
                     sorted_windows = sorted(window_to_keys, reverse=True)
                     selected_window = sorted_windows[1]  # Second most recent
                     tplr.logger.info(f"Selected previous window {selected_window}")
+                elif fetch_previous and len(window_to_keys) <= 1:
```

Comment:
how would you come to this case? line 1522 should protect you?

Reply:
Oh!! you want the `len(window_to_keys) <= 1` case. shoudln't you just change the line 1522 than to be that?

Reply:
So this is a specific case only when we start a run. If we don't have any peers in `comms.peers` we would try and fetch the previous peers (this change was introduced to fix the catch up). However this broke the current setup when we start a completely new run as it would populate it with latest peers. If we change line 1522 to that it wont work when we want the latest peer right? I could however change it to ` len(window_to_keys) == 0 or fetch_previous and len(window_to_keys) <= 1` or something like that. The == 0 we want even if we don't fetch previous peers I assume.

Reply:
no no, it's ok. i think it makes sense. the keyword was that you need the latest peer.